src/Components/ComparePublicationData/ComparePublicationDataModal.tsx

```diff
@@ -132,243 +82,180 @@ const ComparePublicationDataModal: FC<ComparePublicationDataModalProps> = ({
-            ? publicationFromLocalStorage.languages
-            : isDuplicate
-            ? [
-                {
-                  title: state.selectedPublication.title[state.selectedPublication.original_language],
-                  lang: state.selectedPublication.original_language?.toUpperCase(),
-                  original: true,
-                },
-              ]
-            : sortedLanguagesFromImportPublication
-        );
-
-        setSelectedLang(
-          workedOn
-            ? publicationFromLocalStorage?.languages.filter((language: any) => language.original)[0]
-            : isDuplicate
-            ? {
-                title: state.selectedPublication.title[state.selectedPublication.original_language],
-                lang: state.selectedPublication.original_language?.toUpperCase(),
-                original: true,
-              }
-            : importPublication.languages.filter((language: any) => language.original)[0]
-        );
-
-        setAarstall(
-          workedOn
-            ? publicationFromLocalStorage?.yearPublished
-            : isDuplicate
-            ? state.selectedPublication.year_published
-            : importPublication.yearPublished
-        );
-
-        setDoi(
-          workedOn
-            ? publicationFromLocalStorage?.doi
-            : isDuplicate && state.selectedPublication.links
-            ? state.selectedPublication.links[state.selectedPublication.links.length - 1]?.url?.substring(
-                16,
-                state.selectedPublication.links[0]?.url?.length + 1
-              )
-            : importPublication.doi
-            ? importPublication.doi
-            : ''
-        );
-
-        setPublishingDetails(
-          workedOn
-            ? publicationFromLocalStorage?.channel
-            : isDuplicate
-            ? {
-                ...state.selectedPublication.journal,
-                volume: state.selectedPublication.volume || '',
-                pageFrom: state.selectedPublication.pages?.from || '',
-                pageTo: state.selectedPublication.pages?.to || '',
-                issue: state.selectedPublication.issue || '',
-              }
-            : {
-                ...importPublication.channel,
-                volume: importPublication.channel?.volume || '',
-                pageFrom: importPublication.channel?.pageFrom || '',
-                pageTo: importPublication.channel?.pageTo || '',
-                issue: importPublication.channel?.issue || '',
-              }
-        );
       } catch (error) {
         handlePotentialExpiredSession(error);
-        setFetchDataError(error as Error);
+        setLoadContributorsError(error as Error);
+      } finally {
+        setIsLoadingContributors(false);
       }
     }
-
-    setFields().then();
-  }, [isDuplicate, state.selectedPublication, importPublication]);
+    getContributors().then();
+  }, [isDuplicate, state.selectedPublication]);
 
   useEffect(() => {
```

Comment:
Forsøke å slå sammen useEffectene for å unngå re-rendrings loops?

Reply:
gjorde det